### PR TITLE
DDF-2424: Updating registry entries fails when versioning is enabled

### PR DESCRIPTION
**Pull request by rzwiefel**
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard @djblue @tbatie @pklinef 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@coyotesqrl 
@kcwire

### CODE REVIEW

File: catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java [hunk 1]
```diff
@@ -197,8 +198,8 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         //This is a registry store so only allow registry requests through
         if (!filterAdapter.adapt(request.getQuery(),
-                new TagsFilterDelegate(Collections.singleton(RegistryConstants.REGISTRY_TAG),
-                        true))) {
+                new TagsFilterDelegate(Sets.newHashSet(RegistryConstants.REGISTRY_TAG,
```
Comment (coyotesqrl) [inline catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java]: As this was immutable before - `Collections.singleton()` returns an immutable set - we should switch to `ImmutableSet` here.

**Comment by coyotesqrl** [comment]
Other than my mutability comment, this LGTM.

File: catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java (hunk 1 above)
Comment (rzwiefel) [inline catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java]: fixed

**Comment by pklinef** [comment]
👍

**Comment by kcwire** [comment]
👍

**Pull request merged**